bin/hyperlog.js

```diff
@@ -135,93 +135,101 @@ const parseStorage = (dir) => {
-
-    if (cmd === 'read') {
-      if (!key) throw new Error('ERR_KEY_REQUIRED')
-
-      const logConsole = argv.output ? argv.console : true
-      const logFile = argv.output ? fullPath(argv.output) : null
-      const inDir = argv['input-dir']
-        ? new RegExp('^' + escapeRegex(argv['input-dir']))
-        : null
-      const multiFileLog = !!inDir
-
-      if (logFile) {
-        const dirCreated = multiFileLog
-          ? await createDir(logFile)
-          : await createFileDir(logFile)
-        if (!dirCreated) {
-          throw new Error('ERR_MAKE_OUTPUT_DIR_FAILED')
-        }
-      }
+  const argv = yargs.argv
+  const [cmd] = argv._
 
-      let streamOpts = {}
-      if (typeof argv.start === 'number') streamOpts.start = argv.start
-      if (typeof argv.end === 'number') streamOpts.end = argv.end
-      if (argv.tail === true) streamOpts = { snapshot: false, tail: true }
+  if (!cmds.includes(cmd)) throw new Error('ERR_CMD_NOT_SUPPORTED')
 
-      const client = new HyperCoreLogReader(
-        storage, key, null, null, streamOpts
-      )
+  let key = parseKey(argv.key, 64, 'key')
+  let secretKey = parseKey(argv['secret-key'], 128, 'secret-key')
+  const storage = parseStorage(argv.datadir)
 
-      client.on('data', async (data) => {
-        let line = data.toString().trimRight()
-        if (multiFileLog) line = line.replace(inDir, '')
+  if (cmd === 'read') {
+    if (!key) throw new Error('ERR_KEY_REQUIRED')
 
-        if (logConsole) console.log(line)
+    const logConsole = argv.output ? argv.console : true
+    const output = argv.output ? fullPath(argv.output) : null
+    const inDir = argv['input-dir']
+      ? new RegExp('^' + escapeRegex(argv['input-dir']))
+      : null
+    const multiFileLog = !!inDir
+    const destination = output && await prepareOutputDestination(output, multiFileLog)
 
-        if (logFile) {
-          const flags = { flag: 'a' }
-          if (!multiFileLog) {
-            fs.writeFile(logFile, line + '\n', flags, () => { })
-            return
-          }
+    let streamOpts = {}
+    if (typeof argv.start === 'number') streamOpts.start = argv.start
+    if (typeof argv.end === 'number') streamOpts.end = argv.end
+    if (argv.tail === true) streamOpts = { snapshot: false, tail: true }
 
-          const [fpath, ...content] = line.split(' >>> ')
-          const outpath = path.join(logFile, fpath)
-          await createFileDir(outpath)
-          fs.writeFile(outpath, content.join(' >>> ') + '\n', flags, () => { })
-        }
-      })
+    const client = new HyperCoreLogReader(storage, key, null, null, streamOpts)
 
-      await client.start()
-    }
+    client.on('data', async (data) => {
+      let line = data.toString().trimRight()
+      if (multiFileLog) line = line.replace(inDir, '')
 
-    if (cmd === 'write') {
-      if (argv.port && argv.file) throw new Error('ERR_TRANSPORT_AMBIGUOUS')
+      if (logConsole) console.log(line)
```

Comment:
also console log should replace '>>>'

Reply:
should it remove the full path and '>>>'? Currently if I define `--input-dir /foo/bar/`, the console log will print `File-1.log >>> my data`

Reply:
ahh correct now I remember also, since stdout can't demultiplex there we should leave `>>>`